libs/kml/serdes_geojson.cpp

```diff
@@ -0,0 +1,224 @@
+{
+  std::string buffer{};
+  if (glz::write_json(json, buffer))
+    return buffer;
+  else
+    return "<JSON_ERROR>";
+}
+
+bool GeojsonParser::Parse(std::string_view json_content)
+{
+  geojson::GeoJsonData geoJsonData;
+
+  constexpr glz::opts opts{.comments = true, .error_on_unknown_keys = false, .error_on_missing_keys = false};
+  auto ec = glz::read<opts>(geoJsonData, json_content);
+
+  if (ec)
+  {
+    std::string err = glz::format_error(ec, json_content);
+    LOG(LWARNING, ("Error parsing GeoJson:", err));
+    return false;
+  }
+
+  auto getStringFromJsonMap = [](JsonTMap const & propsJson,
+                                 std::string const & fieldName) -> std::optional<std::string>
+  {
+    if (auto const field = propsJson.find(fieldName); field != propsJson.end() && field->second.is_string())
+      return field->second.get_string();
+    return {};
+  };
+
+  // Copy bookmarks from parsed geoJsonData into m_fileData.
+  for (auto & feature : geoJsonData.features)
+  {
+    if (auto const * point = std::get_if<GeoJsonGeometryPoint>(&feature.geometry))
+    {
+      double const longitude = point->coordinates.at(0);
+      double const latitude = point->coordinates.at(1);
+
+      auto const & propsJson = feature.properties;
+      BookmarkData bookmark;
+      bookmark.m_color = ColorData{.m_predefinedColor = PredefinedColor::Red};
+
+      // Parse "name" or "label"
+      if (auto const name = getStringFromJsonMap(propsJson, "name"))
+        kml::SetDefaultStr(bookmark.m_name, *name);
+      else if (auto const label = getStringFromJsonMap(propsJson, "label"))
+        kml::SetDefaultStr(bookmark.m_name, *label);
+
+      // Parse description
+      if (auto const descr = getStringFromJsonMap(propsJson, "description"))
+        kml::SetDefaultStr(bookmark.m_description, *descr);
+
+      // Parse color
+      if (auto const markerColor = getStringFromJsonMap(propsJson, "marker-color"))
+      {
+        auto colorRGBA = ParseHexOsmGarminColor(*markerColor);
+        if (colorRGBA)
+          bookmark.m_color = ColorData{.m_predefinedColor = MapPredefinedColor(*colorRGBA), .m_rgba = *colorRGBA};
+      }
+
+      // Parse icon
+      // if (auto const markerSymbol = getStringFromJsonMap(propsJson, "marker-symbol"))
+      //    bookmark.m_icon = ???;
+      //}
+
+      // UMap custom properties
+      if (auto const umapOptions = propsJson.find("_umap_options");
+          umapOptions != propsJson.end() && umapOptions->second.is_object())
+      {
+        JsonTMap const umap_options = umapOptions->second.get_object();
+        // Parse color from properties['_umap_options']['color']
+        if (auto const color = getStringFromJsonMap(umap_options, "color"))
+        {
+          auto colorRGBA = ParseHexOsmGarminColor(*color);
+          if (colorRGBA)
+            bookmark.m_color = ColorData{.m_rgba = *colorRGBA};
+        }
+
+        // TODO: Store '_umap_options' as a JSON string in some bookmark field.
```

Comment:
Let's make separate issues for this and other unfinished tasks, coordinated by a meta issue GeoJSON support, with links/checkboxes. This one is about interoperability with umap (import/export should not lose properties). Another is about icons support (see above). And there's an unfinished export implementation.

Also there are Polygon, MultiPoint, MultiLineString, and MultiPolygon (polygons should be implemented in KML first; there was already an issue).